system/jhtree/ctfile.cpp

```diff
@@ -324,7 +324,7 @@ bool CWriteNode::add(offset_t pos, const void *indata, size32_t insize, unsigned
         keyPtr += sizeof(rsequence);
         hdr.keyBytes += sizeof(rsequence);
     }
-    if (isLeaf() && (keyType & HTREE_COMPRESSED_KEY))
+    if (isBlob() || (isLeaf() && (keyType & HTREE_COMPRESSED_KEY)))
```

Comment:
I'm pretty sure this test is spurious as writing of blob nodes does not go via this class. I will check.

Reply:
Checked - it was spurious and I have repushed without it